File: api/src/main/java/fr/huiitre/tools/modules/core/notification/application/usecase/DeleteNotificationsUseCase.java
package fr.huiitre.tools.modules.core.notification.application.usecase;

import java.util.List;

import org.springframework.stereotype.Service;

import fr.huiitre.tools.modules.core.role.domain.RoleCode;
import fr.huiitre.tools.modules.core.security.application.ports.CurrentUserProvider;
import fr.huiitre.tools.modules.core.security.application.usecase.SecuredUseCase;
import fr.huiitre.tools.modules.core.notification.application.port.NotificationRepository;

@Service
public class DeleteNotificationsUseCase implements SecuredUseCase {

    private final NotificationRepository notificationRepository;
    private final CurrentUserProvider currentUserProvider;

    public DeleteNotificationsUseCase(NotificationRepository notificationRepository, CurrentUserProvider currentUserProvider) {
        this.notificationRepository = notificationRepository;
        this.currentUserProvider = currentUserProvider;
    }

    @Override
    public RoleCode requiredRole() {
        return RoleCode.READ_ONLY;
    }

    public void execute(List<Long> notificationIds) {
        Long userId = Long.parseLong(currentUserProvider.getCurrentUserId());
        notificationRepository.delete(userId, notificationIds);
    }
}

File: api/Modules/Core/Notifications/Application/Usecases/DeleteNotificationsUseCase.cs
using Tools.Api.Modules.Core.Notifications.Application.Ports;
using Tools.Api.Modules.Core.Security.Application.Services;
using Tools.Api.Modules.Core.Security.Application.Usecases;

namespace Tools.Api.Modules.Core.Notifications.Application.Usecases;

// Supprime des notifications de l'appelant.
//
// Ne disparaît que le lien vers l'utilisateur : le message source reste, les autres destinataires
// le conservent.
public sealed class DeleteNotificationsUseCase(
    UseCaseAuthorizer authorizer,
    INotificationRepository notificationRepository) : SecuredUseCase(authorizer)
{
    // `notificationIds` nul : tout est supprimé.
    public Task Execute(IReadOnlyCollection<long>? notificationIds)
    {
        return notificationRepository.DeleteAsync(CurrentUser.UserId, notificationIds);
    }
}
